libs/application/ui-shell/src/components/Screen.tsx

```diff
@@ -246,7 +269,52 @@ const Screen: FC<ScreenProps> = ({
   useEffect(() => {
     const target = isMobile ? headerHeight : 0
     window.scrollTo(0, target)
-  }, [activeScreenIndex, isMobile])
+
+    if (
+      beforeSubmitCallback.current !== null &&
+      activeScreenIndex !== beforeSubmitCallbackScreenIndex
+    ) {
+      setBeforeSubmitCallback(null)
+    }
```

Comment:
This `beforeSubmitCallbackScreenIndex` is a bit messy.

Is it not safe to always reset the submit callback when activeScreenIndex changes? Or can the new screen already register a new callback before this runs?

Reply:
Yeah you're right, I think it should be safe to refactor. Good catch.